index.js

```diff
@@ -316,4 +344,48 @@ function getBalance(node) {
 	return 0;
 }
 
-//TODO: union, intersection and set difference, helpers, Split and Join
\ No newline at end of file
+// join returns a copy instead of modifying the current trees
+AvlTree.prototype.join = function (element, node2) {
+	var tree1 = this.copy();
+	var tree2 = node2.copy();
+	
```

Comment:
I think you should check that the tree would still be a valid tree here by running the comparison function to confirm that element does indeed fit between the 2 trees.  So you would have to check the key, the min of the right tree, and the max of the left tree, I think.  And throw if it is not.

Reply:
Indeed, for a more robust implementation, this has to be done.